pkgs/servers/nosql/rethinkdb/default.nix

```diff
@@ -5,19 +5,14 @@
 
 stdenv.mkDerivation rec {
   pname = "rethinkdb";
-  version = "2.3.6";
+  version = "2.4.0";
 
   src = fetchurl {
     url = "https://download.rethinkdb.com/dist/${pname}-${version}.tgz";
-    sha256 = "0a6wlgqa2flf87jrp4fq4y9aihwyhgwclmss56z03b8hd5k5j8f4";
+    sha256 = "0k7fm6gz8m62fkd63k81h9bcz4n6vqx6299fyii6fp2r223p1c5z";
   };
 
-  patches = [
-    (fetchurl {
-        url = "https://github.com/rethinkdb/rethinkdb/commit/871bd3705a1f29c4ab07a096d562a4b06231a97c.patch";
-        sha256 = "05nagixlwnq3x7441fhll5vs70pxppbsciw8qjqp660bdb5m4jm1";
-    })
-  ];
+  patches = [ ./v8-no-snapshot.patch ];
```

Comment:
do you mind adding a comment about why this patch is necessary?

Reply:
Will that suffice? I'm not certain how exactly RethinkDB uses V8 or why the snapshots segfault. When I removed the failing patch, which originally disabled V8 snapshots, I noticed that making V8 snapshots would still segfault the build, so I adapted it for the most recent version of RethinkDB, upon which the build succeeded. But I'm unsure I understand what's going on. The commit message of the original patch is "Workaround for building V8 with GCC 6.2".

Reply:
Maybe we could ask @pbogdan? 26ebbac2c42f2d2bb898e9669b2459ff63d0563e

Reply:
The associated PR mentions this issue upstream rethinkdb/rethinkdb#5757, which was closed with mandating `CXX=clang++`.

Reply:
OK, this builds perfectly using `clangStdenv`, so the latest commit switches to that. Let me know if for some reason you prefer using GCC with the patch.